packages/auth/amplify_auth_cognito_android/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/NativeAuthPlugin.kt

```diff
@@ -0,0 +1,410 @@
+import org.json.JSONObject
+
+class NativeAuthPlugin(
+    private val nativeAuthPlugin: () -> NativeAuthPluginBindings.NativeAuthPlugin?
+): AuthPlugin<AWSCognitoAuthServiceBehavior>() {
+
+    override fun getPluginKey(): String = "awsCognitoAuthPlugin"
+
+    override fun getEscapeHatch(): AWSCognitoAuthServiceBehavior? = null
+
+    override fun getVersion(): String = BuildConfig.VERSION_NAME
+
+    override fun configure(pluginConfiguration: JSONObject?, context: Context) {
+        // No-op
+    }
+
+    override fun fetchAuthSession(
+        onSuccess: Consumer<AuthSession>,
+        onError: Consumer<AuthException>
+    ) {
+        val nativePlugin = nativeAuthPlugin()
+        if (nativePlugin == null) {
+            onError.accept(
+                AuthException.UnknownException(
+                    Exception("No native plugin registered")
+                )
+            )
+            return
+        }
+        MainScope().launch {
+            nativePlugin.fetchAuthSession(true) { session ->
+                val couldNotFetchException = AuthException.UnknownException(
+                    Exception("Could not fetch")
+                )
+                val userPoolTokens = if (session.userPoolTokens != null) {
+                    val tokens = AWSCognitoUserPoolTokens(
+                        session.userPoolTokens!!.accessToken,
+                        session.userPoolTokens!!.idToken,
+                        session.userPoolTokens!!.refreshToken,
+                    )
+                    AuthSessionResult.success(tokens)
+                } else {
+                    AuthSessionResult.failure(couldNotFetchException)
+                }
+                val awsCredentials = if (session.awsCredentials != null) {
+                    val sessionCredentials = session.awsCredentials!!
+                    val credentials = Credentials(
+                        sessionCredentials.accessKeyId,
+                        sessionCredentials.secretAccessKey,
+                        sessionCredentials.sessionToken,
+                        if (sessionCredentials.expirationIso8601Utc != null) Instant.fromIso8601(
+                            sessionCredentials.expirationIso8601Utc!!
+                        ) else null,
+                    )
+                    AuthSessionResult.success(credentials)
+                } else {
+                    AuthSessionResult.failure(couldNotFetchException)
+                }
+                val authSession = AWSCognitoAuthSession(
+                    session.isSignedIn,
+                    AuthSessionResult.success(session.identityId),
+                    awsCredentials,
+                    AuthSessionResult.success(session.userSub),
+                    userPoolTokens,
+                )
+                onSuccess.accept(authSession)
+            }
+        }
+    }
+
+    override fun getCurrentUser(): AuthUser {
+        fail("getCurrentUser is not supported")
```

Comment:
nit: Maybe we should use a more helpful message, perhaps conveying why these methods aren't supported (not a huge deal either way though).

Reply:
Done 👍